handlers/handler.go

```diff
@@ -78,14 +80,21 @@ func IndexS3Object(s3objectURL string) {
 	//     <dataguid>/<uuid>/<filename>
 	//
 	// we want to keep the `<dataguid>/<uuid>` part
+	key = strings.Trim(key, "/")
 	split_key := strings.Split(key, "/")
 	var uuid string
-	if len(split_key) == 2 {
+	var filename string
+	foundGUID, errGUID := id.Parse(split_key[0])
+	foundDG, errDG := id.Parse(split_key[1])
```

Comment:
By "DG" do you mean the indexd prefix which usually starts with "dg.xxxx/"? If so, could you refer to "prefix" instead of "dg"? "dg" stands for "dataguid" but the indexd prefix could be any string